docs/dotnet/dotnet-kestrel-metrics.md

```diff
@@ -138,6 +138,50 @@ different processes could be listening on TCP port 12345 and UDP port 12345.
+| `max_concurrent_connections_exceeded` | The connection exceeded the maximum concurrent connection limit. Configured by `KestrelServerLimits.MaxConcurrentConnections`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `max_frame_length_exceeded` | The connection received an HTTP/2 frame that exceeded the size limit specified by `Http2Limits.MaxFrameSize`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `max_request_body_size_exceeded` | The HTTP request body exceeded the maximum request body size limit. Configured by `KestrelServerLimits.MaxRequestBodySize`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `max_request_header_count_exceeded` | The HTTP request headers exceeded the maximum count limit. Configured by `KestrelServerLimits.MaxRequestHeaderCount`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `max_request_headers_total_size_exceeded` | The HTTP request headers exceeded the maximum total size limit. Configured by `KestrelServerLimits.MaxRequestHeadersTotalSize`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `min_request_body_data_rate` | Reading the request body timed out due to data arriving too slowly. Configured by `KestrelServerLimits.MinRequestBodyDataRate`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `min_response_data_rate` | Writing the response timed out because the client did not read it at the specified minimum data rate. Configured by `KestrelServerLimits.MinResponseDataRate`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `missing_stream_end` | The connection received an HTTP/2 `HEADERS` frame for trailers without a stream end flag. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `output_queue_size_exceeded` | The connection exceeded the output queue size and was closed with `INTERNAL_ERROR`. This can be caused by an excessive number of HTTP/2 stream resets. For more information, see [Microsoft Security Advisory CVE-2023-44487:](https://github.com/dotnet/runtime/issues/93303). | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `request_headers_timeout` | Request headers timed out while waiting for headers to be received after the request started. Configured by `KestrelServerLimits.RequestHeadersTimeout`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `response_content_length_mismatch` | The HTTP response body sent data that didn't match the response's `content-length` header. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `server_timeout` | The connection timed out with the `IConnectionTimeoutFeature`. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
Technically correct but not useful. What timeouts signal this?

Reply:
It's from here: https://github.com/dotnet/aspnetcore/blob/b7c8cf7506338b9cd5c4960bd99b03d190c3c805/src/Servers/Kestrel/Core/src/Internal/Infrastructure/TimeoutControl.cs#L312-L324

As far as I can tell nothing in aspnetcore calls `IConnectionTimeoutFeature.SetTimeout`. The API is there for apps to use.

How do you suggest the description is improved?

Reply:
Resolving this, but if there are any suggestions on how to improve things, please bring them on as an issue or PR.